contracts/src/v0.8/dev/automation/2_1/AutomationForwarder.sol

```diff
@@ -21,8 +20,8 @@ contract AutomationForwarder is TypeAndVersionInterface {
 
   error NotAuthorized();
 
-  constructor(uint256 upkeepID, address target) {
-    s_registry = IAutomationRegistryConsumer(msg.sender);
+  constructor(uint256 upkeepID, address target, address registry) {
```

Comment:
curious, why this is needed to be added

Reply:
its just made other code cleaner to directly specify the registry address rather than rely on msg.sender. Ex in migrations I wanted to do the forwarder creation in the transcoder.